pytroll_collectors/scisys.py

```diff
@@ -410,14 +413,25 @@ def receive(self, message):
             if is_uri_on_server(url, strict=True):
                 return self.handle_distrib(url)
         elif message.body.startswith(dispatch_prefix2):
-            filename, arr, url = message.body[len(dispatch_prefix2):].split(" ")
-            url = 'ftp://' + url.replace(':', '')
-            del arr
-            LOGGER.debug("filename = <%s> url = <%s>", filename, url)
-            url = compose_dest_url(filename, url)
-            LOGGER.debug("url = %s", url)
-            if is_uri_on_server(url, strict=True):
+            filename, _, url = message.body[len(dispatch_prefix2):].split(" ")
+            ip_address, path = url.split(":")
+            #import pdb;pdb.set_trace()
+            LOGGER.debug("ip_adress: %s", str(ip_address))
+            LOGGER.debug("self._target_server: %s ", str(self._target_server))
+            if ip_address == self._target_server:
```

Comment:
I suppose that Panu commented on our setup. So this solution is based on our local setup (dispatch_prefix2). May be it would be more important to just check that there is a access to file and it is readable, or what do you think?

Reply:
Yes, but the thing is that we do also receive messages for dispatch starting with `SUCTRN` iirc. So I have to make sure this still works for us. (Sorry for the delay in the answer!)